File: src/lib/software/zulip.py
from src.lib.software import BaseSoftware

class Zulip(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'Zulip-Web-Setup.exe'
        self.silent_install_args = ['/S']

    def resolve_download_url(self):
        self.url_resolved.emit('https://zulip.com/apps/download/windows')

File: packages/app/src/services/software/definitions/zulip.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'zulip';
	public name = 'Zulip';
	public category = [SoftwareCategory.Social];
	public downloadName = 'Zulip-Web-Setup.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'zulip.png';
	public homepage = 'https://zulip.com';

	public async resolveDownloadUrl() {
		return ok('https://zulip.com/apps/download/windows');
	}
}
